site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,995 @@
+    "string": "Computers"
+  }
+}
+```
+
+Both `IS NULL` and `INDEX_IN` will be scalar function expressions. Available 
+functions in Substrait are defined in extension YAML files contained in 
+https://github.com/substrait-io/substrait/tree/main/extensions. Additional
+extensions may be created elsewhere. `IS NULL` is defined as a `is_null` function
+in 
+[functions_comparison.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_comparison.yaml#:~:text=%2D-,name%3A%20%22is_null%22,-description%3A%20Whether)
+and `INDEX_IN` is defined as `index_in` function in
+[functions_set.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_set.yaml#:~:text=%2D-,name%3A%20%22index_in%22,-description%3A%20%3E).
+
+First, the expression for `INDEX_IN("Computers", categories)` is:
+
+<!-- TODO: update once #342 is merged -->
+
+```json
+{
+  "scalarFunction": {
+    "functionReference": 1,
+    "outputType": {
+      "i64": {
+        "nullability": "NULLABILITY_NULLABLE"
+      }
+    },
+    "arguments": [
+      {
+        "enum": {
+          "specified": "NAN_IS_NOT_NAN"
+        }
+      },
+      {
+        "value": {
+          "literal": {
+            "string": "Computers"
+          }
+        }
+      },
+      {
+        "value": {
+          "selection": {
+            "directReference": {
+              "structField": {
+                "field": 1
+              }
+            },
+            "rootReference": {}
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+`functionReference` will be explained later in the plans section. For now,
+understand that it's an index into a list of function definitions and since this
+is the first function we've used it points to the first in that list.
+
+`outputType` defines the type the function outputs. We know this is a nullable
+`i64` type since that is what the function definition declares in the YAML file.
+
+`arguments` defines the arguments being passed into the function, which are all
+done positionally based on the function definition in the YAML file. The first 
+argument is an option that defines how `NaN` equality is handled, and since
+that's not relevant here we just pass `"NAN_IS_NOT_NAN"`. The next two options
+will be familiar as the literal and the field reference we constructed earlier.
+
+To create the final expression, we just need to wrap this in another scalar
+function expression for `IS NULL`.
+
+```json
+{
+  "scalarFunction": {
+    "functionReference": 2,
+    "outputType": {
+      "bool": {
+        "nullability": "NULLABILITY_REQUIRED"
+      }
+    },
+    "arguments": [
+      {
+        "value": {
+          "scalarFunction": {
+            "functionReference": 1,
+            "outputType": {
+              "i64": {
+                "nullability": "NULLABILITY_NULLABLE"
+              }
+            },
+            "arguments": [
+              {
+                "enum": {
+                  "specified": "NAN_IS_NOT_NAN"
```

Comment:
Sorry, but this is out of date (my fault for the breaking change).  Options are no longer treated as arguments but rather their own thing.  So the `NAN_IS_NOT_NAN` will not be set in this way.

Reply:
Yup. Removed those references since they aren't required.